file_optical.c

```diff
@@ -168,10 +168,7 @@ static void fbo_handler_init(struct fbo_handler *h, struct tcmu_device *dev,
 
 static void fbo_handler_destroy(struct fbo_handler *h)
 {
-	if (h->thr) {
-		pthread_kill(h->thr, SIGINT);
-		pthread_join(h->thr, NULL);
-	}
+	tcmulib_cancel_thread(h->thr);
```

Comment:
I'm concerned about using pthread_cancel given that pthread_mutex is used.  If there is any chance of a thread being canceled while holding the mutex, you must register a cleanup handler to release the mutex if there are any thread cancellation points in the critical section.